### PR TITLE
Implement a Simperby client CLI

### PR DESCRIPTION
**Pull request by TomTaehoonKim**

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -15,25 +18,159 @@ fn to_commit_hash(s: &str) -> Result<CommitHash> {
...
+        }) => {
+            let delegation_transaction_data = DelegationTransactionData {
+                delegator: config.public_key,
+                delegatee: serde_spb::from_str(&delegatee)
+                    .map_err(|_| eyre!("invalid delegatee for a delegation transaction"))?,
+                governance,
+                block_height: target_height,
+            };
+            println!(
+                "{:?}",
```
Comment (junha1) [inline cli/src/main.rs:152]: Considering the use of this command, I think it's better to print json

File: cli/src/main.rs [hunk 2]
```diff
@@ -15,25 +18,159 @@ fn to_commit_hash(s: &str) -> Result<CommitHash> {
...
+                .map_err(|_| eyre!("failed to sign"))?
+            );
+        }
+        Commands::Sign(SignCommands::TxUndelegate { target_height }) => {
+            let undelegation_transaction_data = UndelegationTransactionData {
+                delegator: config.public_key,
+                block_height: target_height,
+            };
+            println!(
+                "{:?}",
```
Comment (junha1) [inline cli/src/main.rs:166]: ditto

File: cli/src/main.rs [hunk 3]
```diff
@@ -15,25 +18,159 @@ fn to_commit_hash(s: &str) -> Result<CommitHash> {
...
 }
 
+fn get_timestamp() -> Timestamp {
+    let now = std::time::SystemTime::now();
+    let since_the_epoch = now.duration_since(std::time::UNIX_EPOCH).unwrap();
+    since_the_epoch.as_millis() as Timestamp
+}
+
 async fn run(args: cli::Cli, path: String, config: Config) -> eyre::Result<()> {
     match args.command {
```
[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply cli/src/main.rs:28]: Applied.

File: node/src/lib.rs [hunk 4]
```diff
@@ -136,11 +136,23 @@ pub async fn initialize(config: Config, path: &str) -> Result<SimperbyNode> {
...
 
 /// Runs a server node indefinitely.
-pub async fn serve(_config: Config, _path: &str) -> Result<()> {
-    todo!()
+pub async fn serve(config: Config, path: &str) -> Result<()> {
+    if config.broadcast_interval_ms.is_some() {
+        let node = SimperbyNode::initialize(config.clone(), path).await?;
+        node.serve(
+            config
+                .broadcast_interval_ms
```
[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply node/src/lib.rs]: Applied.

File: node/src/node.rs [hunk 5]
```diff
@@ -183,9 +200,18 @@ impl SimperbyNode {
...
     ) -> Result<()> {
-        unimplemented!()
+        match tx {
+            ExtraAgendaTransaction::Delegate(_) => {
+                self.repository.create_extra_agenda_transaction(&tx).await?;
+            }
+            ExtraAgendaTransaction::Undelegate(_) => {
+                self.repository.create_extra_agenda_transaction(&tx).await?;
+            }
+            ExtraAgendaTransaction::Report(_) => todo!("TxReport is not implemented yet"),
```
[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply node/src/node.rs]: Oops, applied.

File: node/src/node.rs [hunk 6]
```diff
@@ -146,8 +146,25 @@ impl SimperbyNode {
     }
 
     /// Synchronizes the `finalized` branch to the given commit.
```
[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply node/src/node.rs]: Fixed.

File: cli/src/main.rs [hunk 7]
```diff
@@ -46,7 +183,8 @@ async fn run(args: cli::Cli, path: String, config: Config) -> eyre::Result<()> {
                 Signature::sign(hash, &config.private_key).map_err(|_| eyre!("failed to sign"))?
             );
```
[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply cli/src/main.rs:95]: Fixed.

File: cli/src/main.rs (hunk 1 above)
Comment (TomTaehoonKim) [reply cli/src/main.rs:56]: Agree, fixed.

File: cli/src/main.rs (hunk 2 above)
Comment (TomTaehoonKim) [reply cli/src/main.rs:72]: Fixed.

File: cli/src/main.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (junha1) [reply cli/src/main.rs:28]: Use `match` instead of `if`.

[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply cli/src/main.rs:28]: Fixed.

**Review by junha1 — APPROVED** [review]

**Pull request merged**